Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -150,4 +151,105 @@ public static int CountSubstring(this string str, string substring, StringCompar
 
         return count;
     }
+
+    /// <summary>
+    /// Determines if the <paramref name="value"/> is longer than 8 characters or contains an <see cref="Environment.NewLine"/>.
+    /// </summary>
+    public static bool IsLongOrMultiline(this string value)
+    {
+        const int humanReadableLength = 8;
+        return value.Length > humanReadableLength || value.Contains(Environment.NewLine, StringComparison.Ordinal);
+    }
+
+    /// <summary>
+    /// Get the mismatch segment between <paramref name="expected"/> and <paramref name="subject"/> for long strings,
+    /// when they differ at index <paramref name="firstIndexOfMismatch"/>.
+    /// </summary>
+    public static string GetMismatchSegmentForLongStrings(this string subject, string expected, int firstIndexOfMismatch)
+    {
+        int trimStart = CalculateSegmentStart(subject, firstIndexOfMismatch);
+
+        int whiteSpaceCount = (firstIndexOfMismatch - trimStart) + 3;
+        const string prefix = "  ";
+
+        if (trimStart > 0)
+        {
+            whiteSpaceCount++;
+        }
+
+        var visibleText = subject.Substring(trimStart, firstIndexOfMismatch - trimStart);
+        whiteSpaceCount += visibleText.Count(c => c is '\r' or '\n');
+
+        var sb = new StringBuilder();
+
+        sb.Append(' ', whiteSpaceCount).AppendLine("\u2193 (actual)")
```

Comment:
🔧 Use `const` for all usage of unicode characters.

Reply:
Used `const` values for `arrowDown` and `arrowUp` in [9becb00](https://github.com/fluentassertions/fluentassertions/pull/2307/commits/9becb00b59cf8e3fcb7ed2bfdec5b05582b2b44b).